### PR TITLE
contrib/gorilla/mux: option for skipping tracing

### PR DESCRIPTION
**Pull request by stroem**
This update adds an option called WithSkippingFunc, which can be used for skipping tracing based on request path.

Issue #790

### CODE REVIEW

**Review by gbbr — CHANGES_REQUESTED** [review]
Hey! Thanks for contributing to dd-trace-go! Please take a moment to read our short contribution guidelines: https://github.com/DataDog/dd-trace-go/blob/v1/CONTRIBUTING.md

**Comment by stroem** [comment]
> Hey! Thanks for contributing to dd-trace-go! Please take a moment to read our short contribution guidelines: https://github.com/DataDog/dd-trace-go/blob/v1/CONTRIBUTING.md

Wow, you were fast. I was about to add description and link to issue :D

**Comment by stroem** [comment]
@gbbr what about now? :)

**Comment by gbbr** [comment]
Better! Thanks a lot! Usually it's done the other way around and the PR would come last, so I jumped in to look :) Will review soon...

File: contrib/gorilla/mux/mux.go [hunk 1]
```diff
@@ -121,3 +126,7 @@ func defaultResourceNamer(router *Router, req *http.Request) string {
 	}
 	return req.Method + " unknown"
 }
+
+func defaultSkippingFunc(*http.Request) bool {
```
Comment (gbbr) [inline contrib/gorilla/mux/mux.go:130]: I think we can drop this one.

File: contrib/gorilla/mux/option.go [hunk 2]
```diff
@@ -37,6 +38,15 @@ func defaults(cfg *routerConfig) {
 		cfg.serviceName = svc
 	}
 	cfg.resourceNamer = defaultResourceNamer
+	cfg.skippingFunc = defaultSkippingFunc
```
Comment (gbbr) [inline contrib/gorilla/mux/option.go:41]: ... and instead just inline here:
```suggestion
	cfg.ignoreRequest = func(_ *http.Request) { return false }
```

File: contrib/gorilla/mux/mux_test.go [hunk 3]
```diff
@@ -199,6 +199,43 @@ func TestAnalyticsSettings(t *testing.T) {
 	})
 }
 
+func TestSkippingOption(t *testing.T) {
+	skippingFunc := func(req *http.Request) bool {
```
Comment (gbbr) [inline contrib/gorilla/mux/mux_test.go:203]: Can we just inline this into the option of `NewRouter`?

File: contrib/gorilla/mux/mux_test.go [hunk 4]
```diff
@@ -199,6 +199,43 @@ func TestAnalyticsSettings(t *testing.T) {
...
+		{
+			url:       "/skip",
+			spanCount: 0,
+		},
+		{
+			url:       "/200",
+			spanCount: 1,
+		},
+	}
+
```
Comment (gbbr) [inline contrib/gorilla/mux/mux_test.go:220]: Can we remove this line please? I've only recently started to add some [Style guidelines](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#general-guidelines) (check "General Guidelines", point 3) to this repository and specifically tried to mention this part.

File: contrib/gorilla/mux/mux.go (hunk 1 above)
Comment (stroem) [reply contrib/gorilla/mux/mux.go]: Sound good to me

File: contrib/gorilla/mux/mux_test.go (hunk 3 above)
Comment (stroem) [reply contrib/gorilla/mux/mux_test.go]: Sure

File: contrib/gorilla/mux/mux_test.go (hunk 4 above)
Comment (stroem) [reply contrib/gorilla/mux/mux_test.go]: Good style guideline. I think I understand what you were thinking of, are you refer to line 224 aswell?

File: contrib/gorilla/mux/mux.go [hunk 5]
```diff
@@ -94,6 +94,11 @@ func NewRouter(opts ...RouterOption) *Router {
 // We only need to rewrite this function to be able to trace
 // all the incoming requests to the underlying multiplexer
 func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if r.config.ignoreRequest(req) {
+		r.Router.ServeHTTP(w, req)
+		return
+	}
+
```
Comment (gbbr) [inline contrib/gorilla/mux/mux.go:101]: Let's remove this line please 🙏

**Review by gbbr — COMMENTED** [review]
LGTM! One last small thing :)

**Review by knusbaum — APPROVED** [review]
Looks good!

**Comment by knusbaum** [comment]
I want to run unit tests on this before merging.
@stroem if you can upmerge it should run. I'm not sure what happened to cause the lack of unit tests reporting.

**Pull request merged**